waveform_editor/gui/io/manager.py

```diff
@@ -55,27 +58,60 @@ def create_new_file(self):
         self.main_gui.load_yaml(yaml_content)
         self.open_file = None
 
+    def _confirm_and_execute(self, action, message):
+        def proceed():
+            action()
+
+        if self.main_gui.config.has_changed:
+            self.main_gui.confirm_modal.show(message, on_confirm=proceed)
+        else:
+            proceed()
+
     def _handle_menu_selection(self, event):
         clicked = event.new
         if clicked == NEW:
-            self.create_new_file()
+            self._confirm_and_execute(
+                self.create_new_file,
+                (
+                    "### ⚠️ **You have unsaved changes**  \n"
+                    "Opening a new file will discard all unsaved changes.  \n"
+                    "Do you want to continue?"
+                ),
+            )
         elif clicked == OPEN:
-            self.file_loader.open()
+            self._confirm_and_execute(
+                self.file_loader.open,
+                (
+                    "### ⚠️ **You have unsaved changes**  \n"
+                    "Opening a new file will discard all unsaved changes.  \n"
+                    "Do you want to continue?"
+                ),
+            )
         elif clicked == SAVE:
             self.file_saver.save_yaml()
         elif clicked == SAVE_AS:
             self.file_saver.open_save_dialog()
         elif clicked == EXPORT:
-            self.file_exporter.modal.show()
+            self._confirm_and_execute(
+                self.file_exporter.modal.show,
+                (
+                    "### ⚠️ **You have unsaved changes**  \n"
+                    "Exporting now may not include these unsaved changes.  \n"
+                    "Do you want to continue?"
+                ),
+            )
 
-    @param.depends("is_editing", "open_file", watch=True, on_init=True)
+    @param.depends(
+        "is_editing", "open_file", "config.has_changed", watch=True, on_init=True
+    )
     def _set_open_file_text(self):
+        alert = "⚠️ Unsaved changes " if self.main_gui.config.has_changed else ""
         if not self.is_editing:
             self.open_file_text.value = "Create or open a YAML file"
         elif self.open_file:
-            self.open_file_text.value = f"{self.open_file}"
+            self.open_file_text.value = f"{self.open_file}\n{alert}"
         else:
-            self.open_file_text.value = "Untitled_1.yaml"
+            self.open_file_text.value = f"Untitled_1.yaml\n{alert}"
```

Comment:
Can we improve this a bit?
<img width="412" height="79" alt="image" src="https://github.com/user-attachments/assets/a32ba2c5-c705-4136-b336-a8fa3a467b5c" />

1. Can we show the filename only, and maybe a tooltip for displaying the full path?
2. The "unsaved changes" is very present now. Maybe an icon (filled circle?) when there are unsaved changes, and another icon (checkmark?) or no icon at all when there are no unsaved changes? Again with a tooltip to explain the icon.

N.B. I'd be happy with a text field + one tooltip to give more details to cover both.